src/lib/api.js

```diff
@@ -0,0 +1,150 @@
+import axios from 'axios';
+
+const api = {
+    /**
+     * Helper function to convert an Axios error to the format Preview expects
+     *
+     * @private
+     * @param {Object} response - Axios error response
+     * @throws {Error} - Throws when an error response object exists
+     * @return {void}
+     */
+    handleError({ response }) {
+        if (response) {
+            const error = new Error(response.statusText);
+            error.response = response; // Need to pass response through so we can see what kind of HTTP error this was
+            throw error;
+        }
+    },
+
+    /**
+     * Retrieves JSON from response.
+     *
+     * @private
+     * @param {Response} response - Response to parse
+     * @return {Promise|Response} Response if 204, otherwise promise that resolves with JSON
+     */
+    parseResponse: (response) => {
+        if (response.status === 204) {
+            return response;
+        }
+
+        return response.data;
+    },
+
+    /**
+     * Wrapper function for XHR post put and delete
+     *
+     * @private
+     * @param {string} url - URL for XHR
+     * @param {Object} options - Request options
+     * @return {Promise} XHR promise
+     */
+    xhr(url, { data, headers, method, responseType } = {}) {
```

Comment:
Can you please also allow for all requests (event just `get` if it's too much of a hassle) to use `onDownloadProgress`?

This'll allow for us to easily track TTFB